spring-pulsar/src/main/java/org/springframework/pulsar/listener/DefaultPulsarMessageListenerContainer.java

```diff
@@ -296,71 +310,83 @@ public void run() {
 				// Always receive messages in batch mode.
 				try {
 					if (!inRetryMode.get() && !messagesPendingInBatch.get()) {
+						DefaultPulsarMessageListenerContainer.this.receiveInProgress = true;
 						messages = this.consumer.batchReceive();
+						DefaultPulsarMessageListenerContainer.this.receiveInProgress = false;
 					}
 				}
 				catch (PulsarClientException e) {
-					DefaultPulsarMessageListenerContainer.this.logger.error(e, () -> "Error receiving messages.");
-				}
-				Assert.isTrue(messages != null, "Messages cannot be null.");
-				if (this.isBatchListener) {
-					if (!inRetryMode.get() && !messagesPendingInBatch.get()) {
-						messageList = new ArrayList<>();
-						messages.forEach(messageList::add);
+					if (e.getCause() instanceof InterruptedException) {
+						DefaultPulsarMessageListenerContainer.this.logger.debug(e, () -> "Error receiving messages.");
 					}
-					try {
-						if (messageList != null && messageList.size() > 0) {
-							if (this.batchMessageListener instanceof PulsarBatchAcknowledgingMessageListener) {
-								this.batchMessageListener.received(this.consumer, messageList,
-										this.ackMode.equals(AckMode.MANUAL)
-												? new ConsumerBatchAcknowledgment(this.consumer) : null);
-							}
-							else {
-								this.batchMessageListener.received(this.consumer, messageList);
-							}
-							if (this.ackMode.equals(AckMode.BATCH)) {
-								try {
-									if (isSharedSubscriptionType()) {
-										this.consumer.acknowledge(messages);
+					else {
+						DefaultPulsarMessageListenerContainer.this.logger.error(e, () -> "Error receiving messages.");
+					}
+					messages = null;
+				}
+
+				if (messages != null) {
```

Comment:
For readability, I would be in favor of inverting this and short circuiting (continue the loop) rather than indent the large block of logic. Something like 

```java
if (messages == null) {
  continue;
}

if (this.isBatchListener) {
  ...
else {
  ...
}
```
Wdyt?

Reply:
Good idea.